device/pcie/pci_device.cpp

```diff
@@ -190,6 +190,17 @@ inline void memcpy_from_device(void *dest, const void *src, std::size_t num_byte
     }
 }
 
+tt::ARCH PciDeviceInfo::get_arch() const {
+    if (this->device_id == GS_PCIE_DEVICE_ID){
```

Comment:
But we do already have this->arch?

Reply:
I'm thinking of consolidating these kinds of attributes into the `PciDeviceInfo` struct.

By `this` I assume you are referring to a `PCIDevice` instance.  In this case there isn't one.